micro-rdk/src/common/provisioning/storage.rs

```diff
@@ -28,36 +42,68 @@ impl From<CloudConfig> for RobotCredentials {
     }
 }
 
-pub trait CredentialStorage {
+pub trait WifiCredentialStorage {
+    type Error;
+    fn has_wifi_credentials(&self) -> bool;
+    fn store_wifi_credentials(&self, creds: &WifiCredentials) -> Result<(), Self::Error>;
+    fn get_wifi_credentials(&self) -> Result<WifiCredentials, Self::Error>;
+}
+
+pub trait RobotCredentialStorage {
     type Error;
     fn has_stored_credentials(&self) -> bool;
     fn store_robot_credentials(&self, cfg: CloudConfig) -> Result<(), Self::Error>;
     fn get_robot_credentials(&self) -> Result<RobotCredentials, Self::Error>;
 }
 
+#[derive(Default)]
+struct MemoryCredentialStorageInner {
+    config: Option<RobotCredentials>,
+    ssid: Option<String>,
+    pwd: Option<String>,
+}
+
 /// Simple CrendentialStorage made for testing purposes
 #[derive(Default, Clone)]
-pub(crate) struct MemoryCredentialStorage {
-    config: Rc<Mutex<Option<RobotCredentials>>>,
-}
+pub(crate) struct MemoryCredentialStorage(Rc<Mutex<MemoryCredentialStorageInner>>);
```

Comment:
The naming is a little confusing. If `RobotCredentialsStorage` stores robot credentials and `WifiCredentialsStorage` stores wifi credentials then a `MemoryCredentialsStorage` would store ... memory credentials?

Reply:
`InMemoryCredentialStorage`?

Reply:
RAMStorage, god I'am bad at naming stuff :P